### DISCUSSION TITLE
Weekly Blog 2

### DISCUSSION THREAD

**Discussion by LPkkjHD** [SE-TINF22B6/Underwatch#22] [Announcements]
Hello and welcome to the second edition of our weekly blog for the Underwatch Game. This week (as per usual) we have amazing new news to share.

# 🌐 Website
We set up a website that should show all the highscores of the players. We built one landing page and one that actually shows the highscores. The pages have just been initialized, but will be filled with content in the near future.
We build an initial mock-up that shows roughly how the website should look.
![image](https://github.com/SE-TINF22B6/Underwatch/assets/123720714/935816a5-fb75-4ddc-802f-68ba006bf290)
The color scheme and everything else can change, of course.

# 🕹️ Underwatch Game
![image](https://github.com/SE-TINF22B6/Underwatch/assets/11832982/2c1fcfe2-5365-4119-bb07-a2eaf5b55dd3)
We chose the publicly available Assetpack [Pixel_Poem Dungeon Tileset II](https://pixel-poem.itch.io/dungeon-assetpuck) to be the foundation of the game to be created. It features a very pretty set of sprites which we can use in a Editor such as `Tiled` (more on that later) to create TiledMaps which we can load directly into the Game.

As we want Underwatch to be played with decent performance we have to use the GPU and adapt our programming to it. A GPU can work really fast if it has all the data (in our case all images) in the memory stored right on the GPU, however, this only works if we load a single image. Therefore we pack all images into a single giant image and load this into the GPUs memory.

 ![image](https://github.com/SE-TINF22B6/Underwatch/assets/11832982/01b7f008-b7c8-4a30-97b1-91eb4b4caa41) 

Now we can also use the images we can extract as a `TextureRegion` in order to create an Animation. Luckily we don't have to cycle through the `Keyframes` (the images of the animation, like in a thumb cinema) manually and can use a provided feature from the `libGDX` library [Documentation](https://libgdx.com/wiki/graphics/2d/2d-animation).

https://github.com/SE-TINF22B6/Underwatch/assets/11832982/72f71f7b-b146-4486-ab3d-54e2bea77dd3

# 📀 Backend
your content here

**Comment by derrrobin** [comment]
Really nice update. I like the style you choose for your game.  For me it would be quite exciting to get a rough time estimate for your last week. Overall a productive week and i find it very exciting where your game is going. :)

**Reply by LPkkjHD** [reply]
Thank you very much. Our time spent can be tracked on the board of ours [here](https://github.com/orgs/SE-TINF22B6/projects/12/insights/2). This week was on average 6h if I interpret that chart correctly.

**Comment by MaarcS** [comment]
Great blog post! I like the way you used screenshots and text explanations to provide insights of your implementation; especially how you explained your performance challenges without too many technical details.

**Comment by nuo-li** [comment]
Where can I find your SRS draft?

**Reply by LPkkjHD** [reply]
We chose (as per agreement with you) to continue on using the Arc42 Documentation standard which covers most of the SRS. It can be found in our [Wiki](https://github.com/SE-TINF22B6/Underwatch/wiki/System-Architecture-Documentation).

**Comment by TadaXD20** [comment]
The mockup is well made. It is very simple and without any excessive details so it is easy to understand what you want to achieve with this project. 
Did you finish additional tasks than initialization the web pages and creating the mockup? A section with a small summary of the finished tasks would be great for a better overview

Tanina from Team Plapy

**Reply by LPkkjHD** [reply]
Thank you very much for your feedback. To answer the question of yours: Yes we did complete additional tasks apart from the highlights we are presenting here. They can be viewed [here](https://github.com/orgs/SE-TINF22B6/projects/12/views/4).

However, we as a team believe that the weekly blog post should be a fancy way of showing off the results of the past week rather than Copy/Pasting the task board (which is publicly accessible anyways). I hope you can understand that because of the aforementioned reason we are not going to include a task list in the weeks to come.
